SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -72,7 +75,17 @@ void USpatialWorkerConnection::Connect(bool bInitAsClient)
 {
 	if (bIsConnected)
 	{
-		OnConnectionSuccess();
+		AsyncTask(ENamedThreads::GameThread, [WeakThis = TWeakObjectPtr<USpatialWorkerConnection>(this)]
```

Comment:
Why is this now an async task?

Reply:
Not sure to be honest with you. This came from Sahils branch it seems. I think it was added early in the branch history by Joe. I will see if I can work out why we need it.

Reply:
@Vatyx Any idea why this was made Async?

Reply:
Sahil came over and talked to me today. He said that there's some initialization that needs to take place before we call OnConnectionSuccess. The reason the asnyc task was added was to call OnConnectionSuccess on the next frame after the initialization has taken place.